samples/algorithms/ThreeQubitRepetitionCode.qs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Samples {
+
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Measurement;
+
+    @EntryPoint()
+    operation ThreeQubitRepetitionCode() : (Bool, Int) {
+        // Use two qubit registers, one for encoding and an auxiliary one for syndrome measurements.
+        use encodedRegister = Qubit[3];
+        use auxiliaryRegister = Qubit[2];
+
+        // Initialize the first qubit in the register to a |-〉 state.
+        H(encodedRegister[0]);
+        Z(encodedRegister[0]);
+
+        // Apply several unitary operations to the encoded qubits performing bit flip detection and correction between
+        // each application.
+        mutable bitFlipCount = 0;
+        within {
+            // The 3 qubit register is used as a repetition code.
+            Encode(encodedRegister);
+        } apply {
+            let iterations = 5;
+            for _ in 1..iterations {
+                // Apply a sequence of rotations to the encoded register that effectively perform an identity operation.
+                ApplyRotationalIdentity(encodedRegister);
+
+                // Measure the bit flip error syndrome, revert the bit flip if needed, and increase the count if a bit flip occurred.
+                let (parity01, parity12) = MeasureBitFlipSyndrome(encodedRegister, auxiliaryRegister);
+                let bitFlipReverted = RevertBitFlip(encodedRegister, parity01, parity12);
+                if (bitFlipReverted) {
+                    set bitFlipCount += 1;
+                }
+            }
+        }
+
+        // Transform the qubit to the |1〉 state and measure it in the computational basis.
+        H(encodedRegister[0]);
+        let result = MResetZ(encodedRegister[0]) == One;
+        ResetAll(encodedRegister);
+
+        // The output of the program is a boolean-integer tuple where the boolean represents whether the qubit
+        // measurement result was the expected one and the integer represents the number of times bit flips occurred
+        // throughout the program.
+        return (result, bitFlipCount);
+    }
+
+    operation ApplyRotationalIdentity(register : Qubit[]) : Unit is Adj {
+        // This operation implements an identity operation using rotations about the x-axis.
+        // The Rx operation has a period of 2𝜋 (given that it is not possible to measure the difference between
+        // states |Ψ⟩ and -|Ψ⟩). Using it to apply four 𝜋/2 rotations about the x-axis, effectively
+        // leaves the qubit register in its original state.
+        let theta = PI() * 0.5;
+        for i in 1..4 {
+            for qubit in register {
+                Rx(theta, qubit);
+            }
+        }
+    }
+
+    operation RevertBitFlip(register : Qubit[], parity01 : Result, parity12 : Result) : Bool {
+        if parity01 == One {
```

Comment:
This patten of branching seems to be done to avoid the unimplemented LHS dynamic Boolean And support. Rather than tailor the sample to our current intermediate state and then need to remember to come back and clean it up later, we should write the sample the way we intend for it to work and let the generation features catch up. So I'd suggest choosing one of these options:

- Write the if-expressions using And between the conditions that are currently nested, even if that doesn't work right now (it will soon). Worth noting, it won't be as efficient in the QIR produced, but should still work. 
- Convert the two results into a single integer and branch off of that instead.
- Nest using if-else on the first result and have the second parity check inside those branches. This might be the most efficient in terms of QIR instructions produced.

Reply:
I did minimal changes from the original sample just to satisfy the compiler, but I actually like your last suggestion! It makes code cleaner. I updated the code.